### PR TITLE
helper: repo_sync: improve timeout diagnostics

### PR DESCRIPTION
**Pull request by quaresmajose**
- exit after finishing all timeout attempts
- add sleep proportional to the number of attempts before the next sync
- run repo with verbose after some failure

### CODE REVIEW

**Review by doanac — APPROVED** [review]

**Review by vanmaegima — APPROVED** [review]

File: helpers.sh [hunk 1]
```diff
@@ -74,21 +74,29 @@ function repo_sync {
 		status "Adding git config extraheader for $domain/factories"
 		git config --global http.https://${domain}/factories.extraheader "$(cat /secrets/git.http.extraheader)"
 	fi
+	_repo_extra_args = ""
 	for i in $(seq 4); do
-		run repo init --repo-rev=v2.35 --no-clone-bundle -u $* ${REPO_INIT_OVERRIDES} && break
+		run repo init $_repo_extra_args --repo-rev=v2.35 --no-clone-bundle -u $* ${REPO_INIT_OVERRIDES} && break
+		_repo_extra_args = "--verbose"
```
Comment (ricardosalveti) [inline helpers.sh:80]: What is the downside of always running with verbose?

Comment (quaresmajose) [reply helpers.sh:80]: basically it just generates less noise, as the first time it almost always works

File: helpers.sh [hunk 2]
```diff
@@ -74,21 +74,29 @@ function repo_sync {
...
 		status "repo init failed with error $?"
 		[ $i -eq 4 ] && exit 1
 		status "sleeping and trying again"
 		sleep $(($i*2))
 	done
+	_repo_extra_args = ""
 	for i in $(seq 4); do
-		run timeout 4m repo sync && break
+		run timeout 4m repo sync $_repo_extra_args && break
+		_repo_extra_args = "--verbose"
```
Comment (doanac) [inline helpers.sh:89]: clever

**Review by MrCry0 — APPROVED** [review]
LGTM. Cool!

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**